src/jlmkr/utils/dataset.py

```diff
@@ -0,0 +1,186 @@
+# SPDX-FileCopyrightText: © 2024 Jip-Hop and the Jailmakers <https://github.com/Jip-Hop/jailmaker>
+#
+# SPDX-License-Identifier: LGPL-3.0-only
+
+import os.path
+import platform
+import re
+import shutil
+import subprocess
+
+from pathlib import PurePath
+from textwrap import dedent
+
+from paths import JAILS_DIR_PATH, JAIL_CONFIG_NAME, JAIL_ROOTFS_NAME, SCRIPT_DIR_PATH
+from utils.chroot import Chroot
+from utils.console import eprint, YELLOW, BOLD, NORMAL
+
+
+def get_jail_path(jail_name):
+    return os.path.join(JAILS_DIR_PATH, jail_name)
+
+
+def get_jail_config_path(jail_name):
+    return os.path.join(get_jail_path(jail_name), JAIL_CONFIG_NAME)
+
+
+def get_jail_rootfs_path(jail_name):
+    return os.path.join(get_jail_path(jail_name), JAIL_ROOTFS_NAME)
+
+
+def get_relative_path_in_jailmaker_dir(absolute_path):
+    return PurePath(absolute_path).relative_to(SCRIPT_DIR_PATH)
```

Comment:
Maybe instead of `dataset` we could call it `filesystem` for now? It would cover both ZFS datasets as well as plain directories and these other methods may also feel more at home inside a `filesystem` module.